timesketch/lib/datastores/elastic.py

```diff
@@ -518,28 +532,29 @@ def import_event(
             }
 
             if event_id:
+                if event.get(u'lang'):
+                    event = {u'script': event}
+                else:
+                    event = {u'doc': event}
                 header = update_header
-                event = {u'doc': event}
 
             self.import_events.append(header)
             self.import_events.append(event)
             self.import_counter[u'events'] += 1
+
             if self.import_counter[u'events'] % int(flush_interval) == 0:
-                self.client.bulk(
-                    index=index_name,
-                    doc_type=event_type,
-                    body=self.import_events)
+                self.client.bulk(body=self.import_events)
                 self.import_events = []
         else:
             if self.import_events:
-                self.client.bulk(
-                    index=index_name,
-                    doc_type=event_type,
-                    body=self.import_events
-                )
+                self.client.bulk(body=self.import_events)
```

Comment:
Not needed as this is triggered when there is no new events to import. This will flush the queue for any remaining events. I added a comment to make this clear.